examples/systemcalls/systemcalls.c

```diff
@@ -16,7 +21,11 @@ bool do_system(const char *cmd)
  *   and return a boolean true if the system() call completed with success
  *   or false() if it returned a failure
 */
-
+    int ret = system(cmd);
+    if (WIFSIGNALED(ret) && (WTERMSIG (ret) == SIGINT || WTERMSIG (ret) == SIGQUIT))
```

Comment:
@TejaswiniLakshminarayan does this check cover all the conditions listed in the description?  I think you should be using `WIFEXITED` and `WEXITSTATUS` instead.